modeling/algorithm/shared-global-algorithm_test.go

```diff
@@ -0,0 +1,278 @@
+				"ZoneA": types.EndpointSliceGroup{
+					Label: "ZoneA",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneA": types.WeightedEndpoints{Number: 60, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneA": 1,
+					},
+				},
+				"ZoneB": types.EndpointSliceGroup{
+					Label: "ZoneB",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneB": types.WeightedEndpoints{Number: 53, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneB": 1,
+					},
+				},
+				"ZoneC": types.EndpointSliceGroup{
+					Label: "ZoneC",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneC": types.WeightedEndpoints{Number: 55, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneC": 1,
+					},
+				},
+				"global": types.EndpointSliceGroup{
+					Label: "global",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneB": types.WeightedEndpoints{Number: 12, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneA": 0.4,
+						"ZoneB": 0.4,
+						"ZoneC": 0.4,
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+		testCase{
+			name: "mostly balanced large",
+			input: []types.Zone{
+				types.Zone{
+					Nodes:     54,
+					Endpoints: 80,
+					Name:      "ZoneA",
+				},
+				types.Zone{
+					Nodes:     58,
+					Endpoints: 90,
+					Name:      "ZoneB",
+				},
+				types.Zone{
+					Nodes:     74,
+					Endpoints: 105,
+					Name:      "ZoneC",
+				},
+			},
+			expectedOutput: map[string]types.EndpointSliceGroup{
+				"ZoneA": types.EndpointSliceGroup{
+					Label: "ZoneA",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneA": types.WeightedEndpoints{Number: 80, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneA": 1,
+					},
+				},
+				"ZoneB": types.EndpointSliceGroup{
+					Label: "ZoneB",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneB": types.WeightedEndpoints{Number: 80, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneB": 1,
+					},
+				},
+				"ZoneC": types.EndpointSliceGroup{
+					Label: "ZoneC",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneC": types.WeightedEndpoints{Number: 105, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneC": 1,
+					},
+				},
+				"global": types.EndpointSliceGroup{
+					Label: "global",
+					Composition: map[string]types.WeightedEndpoints{
+						"ZoneB": types.WeightedEndpoints{Number: 10, Weight: 1},
+					},
+					ZoneTrafficWeights: map[string]float64{
+						"ZoneA": 0.4,
+						"ZoneB": 0.4,
+						"ZoneC": 0.4,
+					},
+				},
+			},
+			expectedErr: nil,
+		},
+	}
+	localTest := routingAlgorithmTest{
+		algName:   "SharedGlobal",
+		alg:       SharedGlobalAlgorithm{sharedCoreAlgorithm: SharedGlobalAlgorithmCore{globalWeight: 0.4, globalThreshold: 100}},
```

Comment:
Nit: this line seems a bit long, maybe break it up a bit

Reply:
done